serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/servicerecord/ServiceRecordFactory.java

```diff
@@ -59,28 +61,32 @@ public ServiceRecordFactory(ServiceregistryProperties properties, VirkSertServic
         this.krrService = krrService;
     }
 
-    public ServiceRecord createFiksServiceRecord(FiksAdressing fiksAdressing) {
-        return new FiksServiceRecord(fiksAdressing);
+    public Optional<ServiceRecord> createFiksServiceRecord(FiksAdressing fiksAdressing) {
+        if (fiksAdressing.shouldUseFIKS()) {
+            return Optional.of(new FiksServiceRecord(fiksAdressing));
+        }
+        return Optional.empty();
     }
 
     @PreAuthorize("#oauth2.hasScope('move/dpo.read')")
-    public ServiceRecord createEduServiceRecord(String orgnr) {
+    public Optional<ServiceRecord> createEduServiceRecord(String orgnr) {
         Endpoint ep;
         try {
             ep = elmaLookupService.lookup(NORWAY_PREFIX + orgnr);
         } catch (EndpointUrlNotFound endpointUrlNotFound) {
-            if (elmaLookupService.identifierHasInnsynskravCapability(NORWAY_PREFIX + orgnr)) {
-                Audit.info("Does not exist in ELMA with DPO profile, using DPE Innsynskrav",
-                        MarkerFactory.receiverMarker(orgnr));
-                return createDpeServiceRecord(orgnr);
-            } else {
-                logger.info(MarkerFactory.receiverMarker(orgnr),
-                        String.format("Attempted to lookup receiver in ELMA: %s", endpointUrlNotFound.getMessage()));
-                Audit.info("Does not exist in ELMA (no IP?) -> DPV will be used", MarkerFactory.receiverMarker(orgnr));
-                return createPostVirksomhetServiceRecord(orgnr);
-            }
+            logger.info(MarkerFactory.receiverMarker(orgnr),
+                    String.format("Attempted to lookup receiver in ELMA: %s", endpointUrlNotFound.getMessage()));
+            return Optional.empty();
+        }
+
+        String pemCertificate;
+        try {
+            pemCertificate = lookupPemCertificate(orgnr);
+        } catch (CertificateNotFoundException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)